content/blog/understanding-the-dom/index.md

```diff
@@ -0,0 +1,519 @@
+- [The `Element` base class](element-class)
+- [The event system](#events)
+
+## Document Global Object {#document-global-object}
+
+[As mentioned before, the DOM tree must contain one root node](#the-dom). This node, for any instance of the DOM, is the document entry point. When in the browser, this entry point is exposed to the developer with [the global object `document`](https://developer.mozilla.org/en-US/docs/Web/API/Document). This object has various methods and properties to assist in a meaningful way. For example, given a standard HTML5 document:
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+	<title>This is a page title</title>
+</head>
+<body>
+	<p id="mainText">
+    This is the page body
+		<span class="bolded">and it contains</span>
+		a lot of various content within
+    <span class="bolded">the DOM</span>
+	</p>
+</body>
+</html>
+```
+
+The `document` object has the ability to get the `<body>` node ([`document.body`](https://developer.mozilla.org/en-US/docs/Web/API/Document/body)), the `<head>` node ([`document.head`](https://developer.mozilla.org/en-US/docs/Web/API/Document/head)), and even the doctype ([`document.doctype`](https://developer.mozilla.org/en-US/docs/Web/API/Document/doctype)).
+
+![A screenshot of the Chrome debugger console displaying those properties](first_document_properties.png)
+
+### Querying Elements
+
+Additional to containing static references to some of the closest nodes to the root (`<body>` and `<head>`), there is also a way to query for any element by any of the CSS selectors. For example, if we wanted to get a reference to the single element with the `id` of `mainText`, we could use the CSS selector for an id, combined with [the `querySelector` method on the `document`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector):
+
+```javascript
+const mainTextElement = document.querySelector('#mainText');
+```
+
+> The `#` in the `#mainText` is the CSS selector syntax for selecting an element based on it's `id`. If you had a CSS selector of `#testing`, you'd be looking for an element with the following attribute value:
+>
+> ```
+> id="testing"
+> ```
+
+This method will return a reference to the element as rendered in the DOM. [While we'll be covering more of what this reference is able to do later](#element-class), we can do a quick bit of code to show that it's the real element we intended to query:
+
+```javascript
+console.log(mainTextElement.innerHTML); // This will output the HTML that we used to write this element
+```
+
+![A screenshot of the Chrome debugger running the above code](query_selector.png)
+
+We also have the ability to gain a reference to many elements at once. GIven the same HTML document as before, let's say we want to see how many elements have the `bolded` class applied to it. We're able to do so using [the `document` `querySelectorAll` method](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll).
```

Comment:
```suggestion
We also have the ability to gain a reference to many elements at once. Given the same HTML document as before, let's say we want to see how many elements have the `bolded` class applied to it. We're able to do so using [the `querySelectorAll` method on the `document`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll).
```
- For the reference to the method name, I would either make it consistent with the mention to the other method, or I would make it plainly `document.querySelectorAll`.